src/circuit/circuit_frame.rs

```diff
@@ -4698,51 +4721,64 @@ pub fn to_unsigned_int<F: LurkField, CS: ConstraintSystem<F>>(
             &g.power2_64_num,
         )?,
     };
+
     let sum = add(&mut cs.namespace(|| "sum remainder"), &product, &r_num)?;
     let u_decomp = alloc_equal(
         &mut cs.namespace(|| "check unsigned decomposition"),
         &sum,
-        maybe_unsigned,
+        field_elem,
+    )?;
+    enforce_true(
+        &mut cs.namespace(|| "enforce decomposition of unsigned integer"),
+        &u_decomp,
+    )?;
```

Comment:
I think you can combine the `alloc_equal()` and `enforce_true()` with just `equal()`. We could possibly rename to `enforce_equal()` to make that more obvious.

Reply:
done

Reply:
done